Test/Unit/Cron/CatalogIngestionTest.php

```diff
@@ -0,0 +1,144 @@
+ * @magentoDbIsolation disabled
+ */
+class CatalogIngestionTest extends BoltTestCase
+{
+    /**
+     * @var ObjectManagerInterface
+     */
+    private $objectManager;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var ProductEventManager
+     */
+    private $productEventManager;
+
+    /**
+     * @var CatalogIngestionCron
+     */
+    private $catalogIngestionCron;
+
+    /**
+     * @var ProductEventRepositoryInterface
+     */
+    private $productEventRepository;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var ApiHelper|MockObject
+     */
+    private $apiHelper;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUpInternal()
+    {
+        $this->objectManager = Bootstrap::getObjectManager();
+        $this->storeManager = $this->objectManager->get(StoreManagerInterface::class);
+        $this->productEventManager = $this->objectManager->get(ProductEventManager::class);
+        $this->catalogIngestionCron = $this->objectManager->get(CatalogIngestionCron::class);
+        $this->productEventRepository = $this->objectManager->get(ProductEventRepositoryInterface::class);
+        $this->resource = $this->objectManager->get(ResourceConnection::class);
+        $productEventProcessor = $this->objectManager->get(ProductEventProcessor::class);
+        $featureSwitches = $this->createMock(Decider::class);
+        TestHelper::setProperty($this->catalogIngestionCron, 'featureSwitches', $featureSwitches);
+        TestHelper::setProperty($productEventProcessor, 'featureSwitches', $featureSwitches);
+        $featureSwitches->method('isCatalogIngestionEnabled')->willReturn(true);
+        $this->apiHelper = $this->createPartialMock(ApiHelper::class, ['sendRequest']);
+        TestHelper::setProperty($this->productEventManager, 'apiHelper', $this->apiHelper);
+
+        $websiteId = $this->storeManager->getWebsite()->getId();
+        $configData = [
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_ENABLED,
+                'value' => 1,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ]
+        ];
+        TestUtils::setupBoltConfig($configData);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function tearDownInternal(): void
+    {
+        $this->cleanDataBase();
+        parent::tearDownInternal();
+    }
+
+    /**
+     * @test
+     */
+    public function testExecute()
+    {
+        $this->expectExceptionMessage("The bolt product event that was requested doesn't exist.");
+        $product1 = TestUtils::createSimpleProduct();
+        $product2 = TestUtils::createSimpleProduct();
+        $this->apiHelper->expects(self::exactly(2))->method('sendRequest');
+        $this->catalogIngestionCron->execute();
+        $this->productEventRepository->getByProductId($product1->getId());
```

Comment:
not:
1. i'm creating 2 products -> i should have 2 events (i don't check this)
2. i'm running cron expecting 2 calls sendRequest because we have 2 events
3. after success calls bolt events should be removed therefore i'm trying to get them with expected exception:
"The bolt product event that was requested doesn't exist."

Reply:
Right, I misread your code.
Could we refactor it a little? Instead of cathing exceptions via PHPUnit which is a little hard to read we can introduce test helper
```
ifEventExist($id) {
 try {
  $this->productEventRepository->getByProductId($id);
 } catch($e) {
  return false
 }
 return true
}
```